src/textual/drivers/web_driver.py

```diff
@@ -223,6 +241,51 @@ def on_meta(self, packet_type: str, payload: dict[str, object]) -> None:
             self._app.post_message(messages.ExitApp())
         elif packet_type == "exit":
             raise _ExitInput()
+        elif packet_type == "deliver_chunk_request":
+            # A request from the server to deliver another chunk of a file
+            log.debug(f"Deliver chunk request: {payload}")
+            try:
+                delivery_key = cast(str, payload["key"])
+                requested_size = cast(int, payload["size"])
+            except KeyError:
+                log.error("Protocol error: deliver_chunk_request missing key or size")
+                return
+
+            deliveries = self._deliveries
+            deliveries_lock = self._deliveries_lock
+
+            binary_io: BinaryIO | None = None
+            try:
+                with deliveries_lock:
+                    binary_io = deliveries[delivery_key]
+            except KeyError:
+                log.error(
+                    f"Protocol error: deliver_chunk_request invalid key {delivery_key!r}"
+                )
+            else:
+                # Read the requested number of bytes from the file
+                try:
+                    log.debug(f"Reading {requested_size} bytes from {delivery_key}")
+                    chunk = binary_io.read(requested_size)
+                    log.debug(f"Delivering chunk {delivery_key!r} of len {len(chunk)}")
+                    self.write_packed(("deliver_chunk", delivery_key, chunk))
+                    if not chunk:
+                        log.info(f"Delivery complete for {delivery_key}")
+                        binary_io.close()
+                        with deliveries_lock:
+                            del deliveries[delivery_key]
+                except Exception:
+                    binary_io.close()
```

Comment:
This is repeated above. Maybe put this in a `finally:` block ?

Reply:
I only want to close it if there's an exception, not always.